packages/react-composites/src/composites/MeetingComposite/state/MeetingAdapterState.ts

```diff
@@ -71,3 +35,58 @@ export interface MeetingAdapterClientState extends Pick<CallAdapterClientState,
  * combined into one to suit the purpose of a Meeting.
  */
 export interface MeetingAdapterState extends MeetingAdapterUiState, MeetingAdapterClientState {}
+
+export function generateMeetingAdapterState(callAdapter: CallAdapter, chatAdapter: ChatAdapter): MeetingAdapterState {
+  const callAdapterState = callAdapter.getState();
+  const chatAdapterState = chatAdapter.getState();
+
+  const { call, displayName, userId, devices, page } = callAdapterState;
+  const meeting = call ? generateMeetingState(call, chatAdapterState.thread) : undefined;
+
+  return {
+    meeting,
+    userId,
+    page: callPageToMeetingPage(page),
+    displayName,
+    devices
+  };
+}
+
+export function mergeChatAdapterStateIntoMeetingAdapterState(
+  chatAdapterState: ChatState,
+  meetingAdapterState: MeetingAdapterState
+): MeetingAdapterState {
+  const newMeetingState = meetingAdapterState.meeting
+    ? mergeChatStateIntoMeetingState(chatAdapterState.thread, meetingAdapterState.meeting)
+    : undefined;
+
+  const { userId, page, displayName, devices } = meetingAdapterState;
+
+  return {
+    userId,
+    page,
+    displayName,
+    devices,
+    meeting: newMeetingState
+  };
+}
+
+export function mergeCallAdapterStateIntoMeetingAdapterState(
+  callAdapterState: CallAdapterState,
+  meetingAdapterState: MeetingAdapterState
+): MeetingAdapterState {
+  const newMeetingState =
+    meetingAdapterState.meeting && callAdapterState.call
+      ? mergeCallStateIntoMeetingState(callAdapterState.call, meetingAdapterState.meeting)
+      : undefined;
+
+  const { userId, page, displayName, devices } = callAdapterState;
+
+  return {
```

Comment:
Simply `{...callAdapterState, page: callPageToMeetingPage(page), meeting: newMeetingState}`?

Says more clearly exactly what is being overwritten.

Reply:
The reason here is not all callAdapterState is used e.g. callAdapterState has `endedCall` which meetingAdapterState should not have. Using a spread will dump unnecessary props in there also.